src/init.cpp

```diff
@@ -1926,6 +1930,31 @@ bool AppInitMain()
             return InitError(ResolveErrMsg("externalip", strAddr));
     }
 
+    // Read asmap file if configured
+    if (gArgs.IsArgSet("-asmap")) {
+        fs::path asmap_path = fs::path(gArgs.GetArg("-asmap", ""));
+        if (asmap_path.empty()) {
+            asmap_path = DEFAULT_ASMAP_FILENAME;
+        }
+        if (!asmap_path.is_absolute()) {
+            asmap_path = GetDataDir() / asmap_path;
+        }
+        if (!fs::exists(asmap_path)) {
+            InitError(strprintf(_("Could not find asmap file %s"), asmap_path));
+            return false;
+        }
+        std::vector<bool> asmap = CAddrMan::DecodeAsmap(asmap_path);
+        if (asmap.size() == 0) {
+            InitError(strprintf(_("Could not parse asmap file %s"), asmap_path));
+            return false;
+        }
+        const uint256 asmap_version = SerializeHash(asmap);
+        g_connman->SetAsmap(std::move(asmap));
```

Comment:
```suggestion
        node.connman->SetAsmap(asmap);
        const uint256 asmap_version = SerializeHash(asmap);
```

Shouldn't this be like ^ ?

Reply:
We don't use `node.connman` be I've swapped it as asked

Reply:
Right sooo,  what i actually thought it should be while i was opening this suggestion is https://github.com/bitcoin/bitcoin/pull/16702/commits/ec45646de9e62b3d42c85716bfeb06d8f2b507dc#diff-b1e19192258d83199d8adaa5ac31f067af98f63554bfdd679bd8e8073815e69dR1829-R1830
but with `g_connman` instead of `node.connman`(my bad) so: 

```
g_connman->SetAsmap(asmap);
const uint256 asmap_version = SerializeHash(asmap);
```

I wanted to have the `move` dropped because it wasn't in the backport PR and i thought its some custom improvement.

And what do you think now how it should look like @UdjinM6?

Reply:
@xdustinface Given the 18023 PR description I think we should backport it anyway. Backporting PRs in the same order as they were merged originally sounds like a better way but I don't mind backporting it later if you feel like this PR has too many backports already.

Reply:
Ah got it now, im fine adding it here 👍🏻